integration-tests/api/__tests__/admin/analytics-config.js

```diff
@@ -0,0 +1,201 @@
+      const db = useDb()
+      await db.teardown()
+    })
+
+    it("should retrieve config for logged in user if it exists", async () => {
+      // Create config
+      await simpleAnalyticsConfigFactory(dbConnection)
+
+      const api = useApi()
+      const response = await api.get(`/admin/analytics-configs`, adminReqConfig)
+
+      expect(response.data).toMatchSnapshot({
+        analytics_config: {
+          id: expect.any(String),
+          user_id: "admin_user",
+          opt_out: false,
+          anonymize: false,
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+      })
+    })
+
+    it("should return 404 if no config exists", async () => {
+      const api = useApi()
+
+      const err = await api
+        .get(`/admin/analytics-configs`, adminReqConfig)
+        .catch((err) => err)
+
+      expect(err).toBeTruthy()
+      expect(err.response.status).toEqual(404)
+      expect(err.response.data.message).toEqual(
+        "No analytics config found for user with id: admin_user"
+      )
+    })
+  })
+
+  describe("POST /admin/analytics-config", () => {
+    beforeEach(async () => {
+      await adminSeeder(dbConnection)
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      await db.teardown()
+    })
+
+    it("should create a new config for logged in user", async () => {
+      const api = useApi()
+      const response = await api
+        .post(
+          `/admin/analytics-configs`,
+          {
+            opt_out: true,
+            anonymize: true,
+          },
+          adminReqConfig
+        )
+        .catch((e) => console.log(e))
+
+      expect(response.data).toMatchSnapshot({
+        analytics_config: {
+          id: expect.any(String),
+          user_id: "admin_user",
+          opt_out: true,
+          anonymize: true,
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+      })
+    })
+  })
+
+  describe("POST /admin/analytics-config/update", () => {
+    beforeEach(async () => {
+      await adminSeeder(dbConnection)
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      await db.teardown()
+    })
+
+    it("should update the config of the logged in user", async () => {
+      // Create config
+      await simpleAnalyticsConfigFactory(dbConnection)
+
+      const api = useApi()
+      const response = await api.post(
+        `/admin/analytics-configs/update`,
+        {
+          opt_out: true,
+        },
+        adminReqConfig
+      )
+
+      expect(response.data).toMatchSnapshot({
+        analytics_config: {
+          id: expect.any(String),
+          user_id: "admin_user",
+          opt_out: true,
+          anonymize: false,
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+      })
+    })
+
+    it("should fail update the config of the logged in user if a config does not exist", async () => {
```

Comment:
suggestion(non blocking): Maybe we could create the config if it does not exists yet. since it is link to the logged in user it could make sense, and also would make the update idempotent. wdyt?

Reply:
We can definitely do that, it's not something that would ever occur, but I'll add it if we ever switch up how we handle adding and updating these from the admin 😄